core/src/main/java/org/infinispan/LockedStream.java

```diff
@@ -0,0 +1,104 @@
+    * the consumer is given.
+    * @param predicate predicate
+    * @return a LockedStream with the filter applied
+    */
+   LockedStream<K, V> filter(Predicate<? super CacheEntry<K, V>> predicate);
+
+   /**
+    * Same as {@link LockedStream#filter(Predicate)} except that the Predicate must also
+    * implement <code>Serializable</code>
+    * <p>
+    * The compiler will pick this overload for lambda parameters, making them <code>Serializable</code>
+    * @param predicate the predicate to filter out unwanted entries
+    * @return a LockedStream with the filter applied
+    */
+   default LockedStream<K, V> filter(SerializablePredicate<? super CacheEntry<K ,V>> predicate) {
+      return filter((Predicate<? super CacheEntry<K, V>> ) predicate);
+   }
+
+   /**
+    * Performs an action for each element of this stream on the primary owner of the given key.
+    * <p>
+    * This method is performed while holding exclusive lock over the given entry and will be released
+    * only after the consumer has completed. If the entry is directly modified via the
+    * {@link java.util.Map.Entry#setValue(Object)} method this will be the same as if
+    * {@link java.util.Map#put(Object, Object)} was invoked.
+    * <p>
+    * If using a pessimistic transaction this lock is not held using a transaction and thus the user can start a
+    * transaction in this consumer which also must be completed before returning. A transaction can be started in
+    * the consumer and if done it will share the same lock used to obtain the key.
+    * <p>
+    * Some methods on the provided cache may not work as expected. These include
+    * {@link AdvancedCache#putForExternalRead(Object, Object)}, {@link AdvancedCache#lock(Object[])},
+    * {@link AdvancedCache#lock(Collection)}, {@link AdvancedCache#removeGroup(String)}, all of the async methods on
+    * {@link org.infinispan.commons.api.AsyncCache}.
+    * If these methods are used inside of the Consumer on the cache it will throw a {@link IllegalStateException}.
+    * This is due to possible interactions with transactions while using these commands.
+    * @param biConsumer the biConsumer to run for each entry under their lock
+    */
+   void forEach(BiConsumer<Cache<K, V>, ? super CacheEntry<K, V>> biConsumer);
+
+   /**
+    * Same as {@link LockedStream#forEach(BiConsumer)}  except that the BiConsumer must also
+    * implement <code>Serializable</code>
+    * <p>
+    * The compiler will pick this overload for lambda parameters, making them <code>Serializable</code>
+    * @param biConsumer the biConsumer to run for each entry under their lock
+    */
+   default void forEach(SerializableBiConsumer<Cache<K, V>, ? super CacheEntry<K, V>> biConsumer) {
+      forEach((BiConsumer<Cache<K, V>, ? super CacheEntry<K, V>>) biConsumer);
+   }
+
+   /**
+    * Sets the timeout for the acquisition of the lock for each entry.
```

Comment:
Sure, I am thinking I might want to put it on the class definition though instead of the method itself.